src/common/health.rs

```diff
@@ -0,0 +1,104 @@
+use std::collections::HashSet;
+use std::ops::DerefMut as _;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use collection::shards::replica_set::ReplicaState;
+use collection::shards::shard::ShardId;
+use collection::shards::CollectionId;
+use storage::content_manager::toc::TableOfContent;
+use tokio::sync::RwLock;
+
+pub struct Ready {
+    toc: Arc<TableOfContent>,
+    consensus_sync_instant: Instant,
+    initial_unhelathy_shards: RwLock<Option<HashSet<Replica>>>,
+}
+
+impl Ready {
+    pub fn new(toc: Arc<TableOfContent>) -> Self {
+        Self::with_consensus_sync_timeout(toc, Duration::from_secs(10))
+    }
+
+    pub fn with_consensus_sync_timeout(
+        toc: Arc<TableOfContent>,
+        consensus_sync_timeout: Duration,
+    ) -> Self {
+        Self {
+            toc,
+            consensus_sync_instant: Instant::now() + consensus_sync_timeout,
```

Comment:
Should we maybe call this `consensus_warmup` or something along those lines.

Reply:
Nice one, thanks! I just didn't want to spend 15 minutes figuring out the name for the field. 😂

Reply:
I wasn't sure if my suggestion would be either useful or annoying. Happy to see it's the first. :smile: